warehouse/utils/utils.go

```diff
@@ -240,34 +231,32 @@ func GetNewTimings(upload UploadT, dbHandle *sql.DB, status string) []byte {
 	return marshalledTimings
 }
 
-func SetUploadStatus(upload UploadT, status string, dbHandle *sql.DB, additionalFields ...UploadColumnT) (err error) {
-	logger.Infof("WH: Setting status of %s for wh_upload:%v", status, upload.ID)
-	var extraFields string
-	if len(additionalFields) > 0 {
-		for _, f := range additionalFields {
-			extraFields += fmt.Sprintf(`%s=%v,`, f.Column, f.Value)
-		}
-	}
-	marshalledTimings := GetNewTimings(upload, dbHandle, status)
-	sqlStatement := fmt.Sprintf(`UPDATE %s SET status=$1, %s updated_at=$2, timings=$3 WHERE id=$4`, warehouseUploadsTable, extraFields)
-	_, err = dbHandle.Exec(sqlStatement, status, time.Now(), marshalledTimings, upload.ID)
-	if err != nil {
-		panic(err)
-	}
-	return
-}
-
 // SetUploadColumns sets any column values passed as args in UploadColumnT format for warehouseUploadsTable
 func SetUploadColumns(upload UploadT, dbHandle *sql.DB, fields ...UploadColumnT) (err error) {
```

Comment:
My suggestion was not to merge SetUploadStatus and SetUploadColumns. SetUploadColumns should stay as it was, having just the job of updating the table. SetUploadStatus should be rewritten to call SetUploadColumns, after doing its specific tasks (i.e shorthand for status column and computing timings)

Reply:
got it, have both funcs and remvoed repetition of sql query